### PR TITLE
Negative tests

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Need https://github.com/OP-TEE/optee_os/pull/514 to be merge for all tests to pass.

### CODE REVIEW

File: host/xtest/xtest_1000.c [hunk 1]
```diff
@@ -508,6 +509,169 @@ static void xtest_tee_test_1007(ADBG_Case_t *c)
 	TEEC_CloseSession(&session);
 }
 
+static void uuid_to_full_name(char *buf, size_t blen, const TEEC_UUID *uuid,
+			const char *extra_suffix)
+{
+	static const char ta_dir[] = "/lib/teetz";
```
Comment (jbech-linaro) [inline host/xtest/xtest_1000.c]: QEMU (and a few other I guess) are using `/lib/optee_armtz/`

File: host/xtest/xtest_1000.c [hunk 2]
```diff
@@ -577,6 +741,23 @@ static void xtest_tee_test_1008(ADBG_Case_t *c)
 		}
 	}
 	Do_ADBG_EndSubCase(c, "Create session fail");
+
+	Do_ADBG_BeginSubCase(c, "Load fake uuid TA");
+	load_fake_ta(c);
+	Do_ADBG_EndSubCase(c, "Load fake uuid TA");
+
+	Do_ADBG_BeginSubCase(c, "Load corrupt TA");
+	ADBG_EXPECT_TRUE(c, load_corrupt_ta(c,  0, 1)); /* magic */
```
Comment (jbech-linaro) [inline host/xtest/xtest_1000.c]: The `magic`, `img_type`, `img_size` etc will most likely not change in a foreseeable future, but as an alternative and more future proof we could use `offsetof` instead.

**Comment by jbech-linaro** [comment]
Besides my two comments:
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: host/xtest/xtest_1000.c (hunk 2 above)
Comment (jenswi-linaro) [inline host/xtest/xtest_1000.c]: Except that the struct isn't available here. Are you proposing that we should export `signed_hdr.h` in `host_includes`?

File: host/xtest/xtest_1000.c (hunk 1 above)
Comment (jenswi-linaro) [inline host/xtest/xtest_1000.c]: I didn't know that had changed, I'll change it here.
I guess I need to change my test-setup also :-)

File: host/xtest/xtest_1000.c (hunk 2 above)
Comment (jbech-linaro) [inline host/xtest/xtest_1000.c]: Sounds like a good idea to me.

Comment (jenswi-linaro) [inline host/xtest/xtest_1000.c]: OK

**Comment by jenswi-linaro** [comment]
Rebased with tag applied.
Depends on https://github.com/OP-TEE/optee_os/pull/517

File: host/xtest/xtest_1000.c [hunk 3]
```diff
@@ -508,6 +510,169 @@ static void xtest_tee_test_1007(ADBG_Case_t *c)
...
+{
+	char old_buf[PATH_MAX];
+	char new_buf[PATH_MAX];
+
+	uuid_to_full_name(old_buf, sizeof(old_buf), old_uuid, old_extra_suffix);
+	uuid_to_full_name(new_buf, sizeof(new_buf), new_uuid, new_extra_suffix);
+	return !rename(old_buf, new_buf);
+}
+
+static bool copy_file(FILE *src, FILE *dst)
```
Comment (pascal-brand-st-dev) [inline host/xtest/xtest_1000.c]: Can't `rename_file` and `copy_file` have the same prototype?

File: host/xtest/xtest_1000.c [hunk 4]
```diff
@@ -508,6 +510,169 @@ static void xtest_tee_test_1007(ADBG_Case_t *c)
...
+	r = corrupt_file(fsrc, offs, mask);
+	fclose(fsrc);
+
+	if (ADBG_EXPECT_TRUE(c, r)) {
+		res = xtest_teec_open_session(&session,
+					      &create_fail_test_ta_uuid,
+					      NULL, &ret_orig);
+		if (res == TEEC_SUCCESS)
+			TEEC_CloseSession(&session);
+		r |= ADBG_EXPECT_TEEC_RESULT(c, TEEC_ERROR_SECURITY, res);
```
Comment (jforissier) [inline host/xtest/xtest_1000.c]: Shouldn't this be: `r &= ADBG_EXPECT_TEEC_RESULT(c, TEEC_ERROR_SECURITY, res)`?

File: host/xtest/xtest_1000.c [hunk 5]
```diff
@@ -508,6 +510,169 @@ static void xtest_tee_test_1007(ADBG_Case_t *c)
...
+	}
+
+	r |= ADBG_EXPECT_TRUE(c, rename_file(&create_fail_test_ta_uuid, "save",
```
Comment (jforissier) [inline host/xtest/xtest_1000.c]: Same here

File: host/xtest/xtest_1000.c (hunk 4 above)
Comment (jenswi-linaro) [inline host/xtest/xtest_1000.c]: Yes, you're right!

File: host/xtest/xtest_1000.c (hunk 3 above)
Comment (jenswi-linaro) [inline host/xtest/xtest_1000.c]: Yes, I'll fix.

**Comment by jenswi-linaro** [comment]
Comments addressed

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
Tags applied

**Pull request merged**